tests/helper_test.py

```diff
@@ -0,0 +1,235 @@
+
+    def test_ascii_bytes(self):
+        base = u'test'.decode('ascii')
+        expected = u'test'
+        self._test(base, expected)
+
+    def test_utf8_bytes(self):
+        base = u'test'.decode('utf-8')
+        expected = u'test'
+        self._test(base, expected, 'utf-8')
+
+    def test_unicode(self):
+        base = u'test'
+        expected = u'test'
+        self._test(base, expected)
+
+
+class Test_pretty_datetime(unittest.TestCase):
+
+    # TODO: Currently these tests use the ampm format based on whether or not
+    # the testing machine's locale sets them. To be really good mock should be
+    # used to change the locale between an am/pm locale and a 24 hour locale
+    # and test both scenarios.
+
+    __patchers = []
+
+    @classmethod
+    def setUpClass(cls):
+        # Create a random number generator, but seed it so that it will produce
+        # deterministic output. This is used to select a subset of possible
+        # values for each of the tests in this class, since otherwise they
+        # would get really expensive (time wise).
+        cls.random = random.Random()
+        cls.random.seed(42)
+
+        # Pick an exact date to ensure that the tests run the same no matter
+        # what time of day they're run.
+        cls.now = datetime.datetime(2000, 1, 5, 12, 0, 0, 0)
+
+        # Mock datetime.now, which ensures that the time is always the same
+        # removing race conditions from the tests.
+        dt = mock.Mock()
+        dt.now = mock.Mock(return_value=cls.now)
+        cls.__patchers.append(mock.patch('alot.helper.datetime', dt))
+
+        for p in cls.__patchers:
+            p.start()
+
+    @classmethod
+    def tearDownClass(cls):
+        for p in cls.__patchers:
+            p.stop()
+
+    def test_just_now(self):
+        for i in (self.random.randint(0, 60) for _ in xrange(5)):
+            test = self.now - datetime.timedelta(seconds=i)
+            actual = helper.pretty_datetime(test)
+            self.assertEquals(actual, u'just now')
+
+    def test_x_minutes_ago(self):
+        for i in (self.random.randint(60, 3600) for _ in xrange(10)):
+            test = self.now - datetime.timedelta(seconds=i)
+            actual = helper.pretty_datetime(test)
+            self.assertEquals(
+                actual, u'{}min ago'.format((self.now - test).seconds // 60))
+
+    def test_x_hours_ago(self):
+        for i in (self.random.randint(3600, 3600 * 6) for _ in xrange(10)):
+            test = self.now - datetime.timedelta(seconds=i)
+            actual = helper.pretty_datetime(test)
+            self.assertEquals(
+                actual, u'{}h ago'.format((self.now - test).seconds // 3600))
+
+    # TODO: yesterday
+    # TODO: yesterday > now > a year
+    # TODO: last year
+    # XXX: when can the last else be hit?
+
+    @staticmethod
+    def __future_expected(test):
+        if test.strftime('%P'):
+            expected = test.strftime('%I:%M%P')
```

Comment:
It might be better to not merge the tests for 'pretty_datetime' just yet, that one is only half-baked. I'll send an updated pull request with that test removed, and I'll clean it up for later. Apart from that problem (which was copied from alot) there are a number of cases marked as TODO, and a few other bits that need to be fixed.